pymc/model_graph.py

```diff
@@ -302,164 +348,215 @@ def get_plates(self, var_names: Iterable[VarName] | None = None) -> dict[str, se
+        ]
 
-        node_formatters = node_formatters or {}
-        node_formatters = update_node_formatters(node_formatters)
-
-        graph = graphviz.Digraph(self.model.name)
-        for plate_label, all_var_names in self.get_plates(var_names).items():
-            if plate_label:
-                # must be preceded by 'cluster' to get a box around it
-                with graph.subgraph(name="cluster" + plate_label) as sub:
-                    for var_name in all_var_names:
-                        self._make_node(
-                            var_name, sub, formatting=formatting, node_formatters=node_formatters
-                        )
-                    # plate label goes bottom right
-                    sub.attr(label=plate_label, labeljust="r", labelloc="b", style="rounded")
-            else:
-                for var_name in all_var_names:
-                    self._make_node(
-                        var_name, graph, formatting=formatting, node_formatters=node_formatters
-                    )
 
-        for child, parents in self.make_compute_graph(var_names=var_names).items():
-            # parents is a set of rv names that precede child rv nodes
-            for parent in parents:
-                graph.edge(parent.replace(":", "&"), child.replace(":", "&"))
+def make_graph(
+    name: str,
+    plates: list[Plate],
+    edges: list[tuple[VarName, VarName]],
+    formatting: str = "plain",
+    save=None,
+    figsize=None,
+    dpi=300,
+    node_formatters: NodeTypeFormatterMapping | None = None,
+    include_shape_size: bool = True,
+):
+    """Make graphviz Digraph of PyMC model
 
-        if save is not None:
-            width, height = (None, None) if figsize is None else figsize
-            base, ext = path.splitext(save)
-            if ext:
-                ext = ext.replace(".", "")
-            else:
-                ext = "png"
-            graph_c = graph.copy()
-            graph_c.graph_attr.update(size=f"{width},{height}!")
-            graph_c.graph_attr.update(dpi=str(dpi))
-            graph_c.render(filename=base, format=ext, cleanup=True)
+    Returns
+    -------
+    graphviz.Digraph
+    """
+    try:
+        import graphviz
+    except ImportError:
+        raise ImportError(
+            "This function requires the python library graphviz, along with binaries. "
+            "The easiest way to install all of this is by running\n\n"
+            "\tconda install -c conda-forge python-graphviz"
+        )
 
-        return graph
+    node_formatters = node_formatters or {}
+    node_formatters = update_node_formatters(node_formatters)
+
+    graph = graphviz.Digraph(name)
+    for plate in plates:
+        if plate.meta:
+            # must be preceded by 'cluster' to get a box around it
+            plate_label = create_plate_label(
+                plate.variables[0].var.name,
+                plate.meta,
+                include_size=include_shape_size,
+            )
```

Comment:
Should `create_plate_label` now take `plate_formatters` that among other things decides on whether to include_size?

Reply:
Yeah, I think that is fair. Where do you view that being exposed?

Reply:
I exposed the create_plate_label in both `make_graph` and `make_networkx`. However, left it out in the `model_to_graphviz` and `model_to_networkx` functions.

If a user defines `Callable[[DimInfo], str]` function, then that can be used in the first two, more general functions